hpx/util/tuple.hpp

```diff
@@ -856,16 +833,16 @@ namespace hpx { namespace util
             detail::pack_c<std::size_t, Is...>, detail::pack<Tuples...>
         >
         {
-            typedef tuple<
-                typename tuple_cat_element<Is, detail::pack<Tuples...> >::type...
-            > type;
```

Comment:
With this last bit gone, there doesn't seem to be a reason anymore to keep `tuple_cat_result_impl` as a helper struct rather than a helper function.